src/Http/Controllers/ActionController.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace HeadlessLaravel\Formations\Http\Controllers;
+
+use Illuminate\Bus\Batch;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Bus;
+use Throwable;
+
+class ActionController extends Controller
+{
+    public function store(Request $request)
+    {
+        $currentAction = $this->formationAction();
+
+        if (!$currentAction) {
+            return redirect()->route('formation.index');
+        }
+
+        $modelIds = $request->get('selected', []);
+
+        $models = [];
+        if ($modelIds === 'all') {
+            $models = $currentAction->formation->results();
+        } elseif (is_int($modelIds)) {
+            $model = $this->model()->where($this->model()->getKeyName(), $modelIds)->first();
+            if (!empty($model)) {
+                $models = [$model];
+            }
+        } elseif (is_array($modelIds)) {
+            $models = $this->model()->whereIn($this->model()->getKeyName(), $modelIds)->get();
+        }
+
+        $jobs = [];
+        foreach ($models as $model) {
+            $jobs[] = new $currentAction->job($model, $currentAction->fields);
+        }
+
+        $batch = Bus::batch($jobs)->then(function (Batch $batch) {
+            // All jobs completed successfully...
+        })->catch(function (Batch $batch, Throwable $e) {
+            // First batch job failure detected...
+        })->finally(function (Batch $batch) {
+            // The batch has finished executing...
+        })->dispatch();
+
```

Comment:
@dillingham I'm not sure how we will get `batchId` if we move `Bus::batch` and dispatch function to the `BulkAction` Job.
Because then batch will be created in Job and which will happen in `BulkAction` Job.

Reply:
You're right, I read this article from spatie and they handle it by just using cursor() in the solving memory issues section 

https://freek.dev/1734-how-to-group-queued-jobs-using-laravel-8s-new-batch-class

```php
$batch = Bus::batch([])
    ->allowFailures()
    ->dispatch()

$subscribersQuery
    ->cursor()
    ->map(fn (Subscriber $subscriber) => $this->createSendMailJob($campaign, $subscriber, $segment))
    ->filter()
    ->chunk(1000)
    ->each(function (LazyCollection $jobs) use ($batch) {
        $batch->add($jobs); // 1000 jobs are now added in one go
    });

Reply:
Done